datanode/datanode.go

```diff
@@ -469,7 +474,18 @@ func (d *dataNode) AssignShardSet(shardSet shard.ShardSet) {
 			}
 		}()
 	}
-	d.RUnlock()
+}
+
+// GetOwnedShards returns all shard ids the datanode owns
+func (d *dataNode) GetOwnedShards() []int {
```

Comment:
kind of duplicate as above (t *topologyShardOwner) GetOwnedShards, can we use the d.topo directkly?

Reply:
no topologyShardOwner gets all shards from the topology

this get the shards this host owns